dolphinscheduler-dao/src/main/resources/sql/upgrade/3.1.0_schema/postgresql/dolphinscheduler_ddl.sql

```diff
@@ -15,44 +15,52 @@
 
-CREATE OR REPLACE FUNCTION public.dolphin_update_metadata(
-    )
-    RETURNS character varying
-    LANGUAGE 'plpgsql'
-    COST 100
-    VOLATILE PARALLEL UNSAFE
-AS $BODY$
-DECLARE
-v_schema varchar;
-BEGIN
-    ---get schema name
-    v_schema =current_schema();
-    
-ALTER TABLE t_ds_project alter COLUMN description type varchar(255);
-ALTER TABLE t_ds_task_group alter COLUMN description type varchar(255);
-
---- add column
-EXECUTE 'ALTER TABLE ' || quote_ident(v_schema) ||'.t_ds_task_definition ADD COLUMN IF NOT EXISTS task_execute_type int DEFAULT ''0''  ';
-EXECUTE 'ALTER TABLE ' || quote_ident(v_schema) ||'.t_ds_task_definition_log ADD COLUMN IF NOT EXISTS task_execute_type int DEFAULT ''0''  ';
-EXECUTE 'ALTER TABLE ' || quote_ident(v_schema) ||'.t_ds_task_instance ADD COLUMN IF NOT EXISTS task_execute_type int DEFAULT ''0''  ';
-EXECUTE 'ALTER TABLE ' || quote_ident(v_schema) ||'.t_ds_task_instance ADD COLUMN IF NOT EXISTS task_execute_type int DEFAULT ''0''  ';
-EXECUTE 'ALTER TABLE ' || quote_ident(v_schema) ||'.t_ds_task_instance DROP CONSTRAINT foreign_key_instance_id';
-
-
-return 'Success!';
-exception when others then
-        ---Raise EXCEPTION '(%)',SQLERRM;
+-- t_ds_task_definition_log
+ALTER TABLE t_ds_task_instance ADD COLUMN IF NOT EXISTS cpu_quota int NOT NULL DEFAULT '-1';
+ALTER TABLE t_ds_task_instance ADD COLUMN IF NOT EXISTS memory_max int NOT NULL DEFAULT '-1';
 
-        return SQLERRM;
-END;
-$BODY$;
+-- t_ds_relation_process_instance
+DROP INDEX IF EXISTS "idx_relation_process_instance_parent_process_task";
+CREATE INDEX IF NOT EXISTS idx_relation_process_instance_parent_process_task ON t_ds_relation_process_instance USING Btree("parent_process_instance_id","parent_task_instance_id");
+DROP INDEX IF EXISTS "idx_relation_process_instance_process_instance_id";
+CREATE INDEX IF NOT EXISTS idx_relation_process_instance_process_instance_id ON t_ds_relation_process_instance USING Btree("process_instance_id");
 
-select dolphin_update_metadata();
-
-
-d//
+-- t_ds_cluster
+CREATE TABLE IF NOT EXISTS "t_ds_cluster" (
+    id serial NOT NULL,
+    code bigint NOT NULL,
+    name varchar(100) DEFAULT NULL,
+    config text DEFAULT NULL,
+    description text,
+    operator int DEFAULT NULL,
+    create_time timestamp DEFAULT NULL,
+    update_time timestamp DEFAULT NULL,
+    PRIMARY KEY (id),
+    CONSTRAINT cluster_name_unique UNIQUE (name),
+    CONSTRAINT cluster_code_unique UNIQUE (code)
+);
 
+--- add column
+ALTER TABLE t_ds_task_definition ADD COLUMN IF NOT EXISTS task_execute_type int DEFAULT '0';
+ALTER TABLE t_ds_task_definition_log ADD COLUMN IF NOT EXISTS task_execute_type int DEFAULT '0';
+ALTER TABLE t_ds_task_instance ADD COLUMN IF NOT EXISTS task_execute_type int DEFAULT '0';
+ALTER TABLE t_ds_task_instance DROP CONSTRAINT foreign_key_instance_id;
+ALTER TABLE t_ds_project alter COLUMN description type varchar(255);
+ALTER TABLE t_ds_task_group alter COLUMN description type varchar(255);
```

Comment:
3.1.0 postgre can native re-run with `if exists` or `if not eixsts`, so we I using SQL for it